homeassistant/components/samsungtv/bridge.py

```diff
@@ -0,0 +1,241 @@
+        ):
+            # We got a response so it's working.
+            return STATE_ON
+        except OSError:
+            # Different reasons, e.g. hostname not resolveable
+            return STATE_OFF
+
+        return STATE_OFF
+
+    def _get_remote(self):
+        """Create or return a remote control instance."""
+        if self._remote is None:
+            # We need to create a new instance to reconnect.
+            try:
+                LOGGER.debug("Create SamsungRemote")
+                self._remote = Remote(self.config.copy())
+            # This is only happening when the auth was switched to DENY
+            # A removed auth will lead to socket timeout because waiting for auth popup is just an open socket
+            except AccessDenied:
+                self._notify_callback()
+                raise
+        return self._remote
+
+    def _send_key(self, key):
+        """Send the key using legacy protocol."""
+        self._get_remote().control(key)
+
+
+class SamsungTVWSBridge(SamsungTVBridge):
+    """The Bridge for WebSocket TVs."""
+
+    def __init__(self, config):
+        """Initialize Bridge."""
+        super().__init__(config)
+        self.config = config
+
+    def try_connect(self, port):
+        """Try to connect to the Websocket TV."""
+        for self.port in (8001, 8002):
+            if port is None or port == self.port:
+                config = {
+                    "name": "HomeAssistant",
+                    "description": "HomeAssistant",
+                    "host": self.host,
+                    "method": self.method,
+                    "port": self.port,
+                    # We need this high timeout because waiting for auth popup is just an open socket
+                    "timeout": 31,
+                    "token": self.token,
+                }
+                try:
+                    LOGGER.debug("Try config: %s", config)
+                    with SamsungTVWS(
+                        host=self.host,
+                        port=self.port,
+                        token=self.token,
+                        timeout=config["timeout"],
+                        name=config["name"],
+                    ) as remote:
+                        remote.open()
+                    LOGGER.debug("Working config: %s", config)
+                    LOGGER.debug("Token %s", self.token)
+                    return RESULT_SUCCESS
+                except WebSocketException:
+                    LOGGER.debug("Working but unsupported config: %s", config)
+                    return RESULT_NOT_SUPPORTED
+                except (OSError, Exception) as err:  # pylint: disable=broad-except
+                    LOGGER.debug("Failing config: %s, error: %s", config, err)
+
+        return RESULT_NOT_SUCCESSFUL
+
+    def get_state(self):
+        """Get TV state."""
+        try:
+            ping_url = f"http://{self.host}:8001/api/v2/"
+
+            requests.get(ping_url, timeout=1)
```

Comment:
We could have a small speed up with `requests.head(`.

Reply:
very small but you're right

Reply:
This isn't allowed in home assistant. It's needs to be moved to a standalone library.

Please remove it.

Reply:
@MartinHjelmare do you mean the direct requests call, don't you?.
I wasn't aware... No problem, I'll add a method on samsungtvws.

Reply:
Yes.

Reply:
I found that the recent 1.4.0 version of samsungtvws already added a method that calls the same URL. I used it but at the end I found that the resulting code would have been very similar to the one used for the samsung legacy TVs... 
So I changed my mind and I choosed to return to the "original" update code.